crates/query-engine/execution/src/mutation.rs

```diff
@@ -114,17 +114,24 @@ async fn execute_query(
         .await?;
 
     // run and fetch from the database
-    sqlx_query
+    let check_constraint = sqlx_query
         .try_map(|row: sqlx::postgres::PgRow| {
-            let mut bytes = row.try_get_raw(0)?.as_bytes().unwrap();
+            let mut results_bytes = row.try_get_raw(0)?.as_bytes().unwrap();
             // If the result is JSONB, PostgreSQL adds a 0x01 at the start of
             // the buffer, which we need to explicitly discard.
             // This will never be valid JSON, so we can discard it safely.
-            if bytes.first() == Some(&1) {
-                bytes = &bytes[1..];
+            if results_bytes.first() == Some(&1) {
+                results_bytes = &results_bytes[1..];
+            }
+            buffer.put(results_bytes);
+
+            // if our mutation also includes a check constraint, return it.
+            if row.len() == 2 {
+                let check_constraint = row.try_get::<bool, usize>(1)?;
+                Ok(Some(check_constraint))
+            } else {
+                Ok(None)
             }
```

Comment:
fetch the check constraint field.

Reply:
Is there only ever one row returned here? Do we not do `variable` / `foreach` for mutations?